src/go/vtunnel/pkg/vmsock/conn_linux.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright © 2022 SUSE LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vmsock
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/linuxkit/virtsock/pkg/vsock"
+	"github.com/sirupsen/logrus"
+
+	"github.com/rancher-sandbox/rancher-desktop/src/go/wsl-helper/pkg/dockerproxy/util"
+)
+
+type PeerConnector struct {
+	IPv4ListenAddress  string
+	TCPListenPort      int
+	VsockHandshakePort uint32
+	VsockHostPort      uint32
+}
+
+// ListendAndHandshake listens for incoming VSOCK connections from the Host process
+// The handshake is perfomed once during startup/restart to make sure that
+// host process is talking to a right hyper-v VM (most likely WSL)
+func (p *PeerConnector) ListendAndHandshake() {
+	l, err := vsock.Listen(vsock.CIDAny, p.VsockHandshakePort)
+	if err != nil {
+		logrus.Fatalf("PeerHandshake listen for incoming vsock: %v", err)
+	}
+	defer l.Close()
+
+	for {
+		conn, err := l.Accept()
+		if err != nil {
+			logrus.Errorf("PeerHandshake accepting incoming socket connection: %v", err)
+		}
+		_, err = conn.Write([]byte(SeedPhrase))
+		if err != nil {
+			logrus.Errorf("PeerHandshake writing seed phrase: %v", err)
+		}
+
+		conn.Close()
+		logrus.Info("successful handshake with vsock-host")
```

Comment:
Yes, because the for loop is an ongoing thing, for example, the host process can stop/start/restart, etc, so the peer is always listening for handshake requests.